File: domain/src/main/java/com/github/anrimian/musicplayer/domain/controllers/SystemServiceController.java
package com.github.anrimian.musicplayer.domain.controllers;

import io.reactivex.rxjava3.core.Observable;

public interface SystemServiceController {

    void startMusicService();

    default void stopMusicService() {
        stopMusicService(false);
    }

    void stopMusicService(boolean forceStop);

    Observable<Object> getStopForegroundSignal();

}

File: domain/src/main/java/com/github/anrimian/musicplayer/domain/controllers/SystemServiceController.kt
package com.github.anrimian.musicplayer.domain.controllers

import io.reactivex.rxjava3.core.Observable

interface SystemServiceController {

    fun startMusicService()

    fun stopMusicService(forceStop: Boolean = false, hideUi: Boolean = false)

    fun getStopForegroundSignal(): Observable<Boolean>

}